### PR TITLE
Rework telemetry; add README for how to use relative paths

### PR DESCRIPTION
**Pull request by msarahan**

### CODE REVIEW

File: github-actions-job-info/action.yml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+          var this_job = undefined;
+          var retryCount = 0;
+          maxRetries = 5;
+          initialDelay = 2000;
+          while (this_job === undefined) {
+            this_job = await get_job({github, context, process});
+            if (this_job === undefined && retryCount < maxRetries) {
+              retryCount++;
+              const delay = initialDelay * Math.pow(2, retryCount - 1);
```
[the remark this answers is not in the document]
Comment (msarahan) [reply github-actions-job-info/action.yml:43]: where's the dumpster fire emoji when you need it?

**Comment by msarahan** [comment]
This PR depends on https://github.com/rapidsai/ci-imgs/pull/196 being merged and all images built/pushed. Removing [`9016654` (#20)](https://github.com/rapidsai/shared-actions/pull/20/commits/90166545a7f7b80efa0dde65acaf7de06562b62f) would alleviate that.

File: telemetry-finalize-root-span/action.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+runs:
+  using: 'composite'
+  steps:
+    - name: Get final timestamp
+      id: timestamp
+      run:
+          echo "FINAL_TIME=$(date +%s.%N)" >> ${GITHUB_OUTPUT}
+    - name: Update root span with final completion time
+      if: always()
+      uses: rapidsai/shared-actions/telemetry-create-span@add-telemetry
```
Comment (ajschmidt8) [inline telemetry-finalize-root-span/action.yml:31]: should this be a relative path?

File: telemetry-summarize/action.yml [hunk 3]
```diff
@@ -1,110 +1,108 @@
...
+        export TRACEPARENT=${job_traceparent}
+
+        # The reporting of the completion time is earlier than the actual last step's completion.
+        # We compensate for that by picking up the last known completion time of any step.
+        last_timestamp=$(date +'%s')
+
+        # Base64 encoding is to avoid issues with spaces/newlines/whatever funny business
+        for row in $(cat job_info.json | jq -r '.steps[] | @base64'); do
+            name="$(echo ${row} | base64 --decode | jq -r ".name")"
+            conclusion="$(echo ${row} | base64 --decode | jq -r ".conclusion")"
```
Comment (ajschmidt8) [inline telemetry-summarize/action.yml:46]: Just wanted to point out that `jq` has a `@base64d` function as well: https://jqlang.github.io/jq/manual/#format-strings-and-escaping

image

Makes no differences to me whether you change these lines, just wanted to mention it.

**Review by ajschmidt8 — APPROVED** [review]

**Pull request merged**